d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -440,6 +471,74 @@ private void getChildrenData(String path, Collection<String> children, Callback<
     }
   }
 
+  private void setRawClientTrackingNode()
+  {
+    String rawD2ClientTrackingPath = D2Utils.getRawClientTrackingPath();
+    try
+    {
+      if (_zk.exists(rawD2ClientTrackingPath, false) != null)
+      {
+        LOG.debug("rawClientTracking node already exist at path: {}", rawD2ClientTrackingPath);
+        return;
+      }
+
+      List<String> childNodes = _zk.getChildren(D2Utils.getRawClientTrackingBasePath(), false);
+      // If tracking node count exceeds the max limit, we won't create further Znode.
+      if (childNodes.size() > D2Utils.RAW_D2_CLIENT_MAX_TRACKING_NODE)
+      {
+        LOG.error("The number of rawD2ClientBuilders exceeds the limit: {}, skipping further node creation for RawClientTracking", childNodes.size());
```

Comment:
nit: could be clearer about that it's about the znode count under what path, such as 
The number of znodes under {} exceeds the limit: {}, ......", rawD2ClientTrackingPath, childNodes.size()